communication/mavlink_waypoint_handler.cpp

```diff
@@ -297,7 +297,15 @@ static void waypoint_handler_control_time_out_waypoint_msg(mavlink_waypoint_hand
  *
  * \return  The waypoint in local coordinate frame
  */
-static local_position_t waypoint_handler_set_waypoint_from_frame(waypoint_struct_t* current_waypoint, global_position_t origin);
+static waypoint_local_struct_t waypoint_handler_set_waypoint_from_frame(waypoint_struct_t* current_waypoint, global_position_t origin, dubin_state_t* dubin_state);
+
+/**
+ * \brief   Computes the state machine for the Dubin navigation type
+ *
+ * \param   waypoint_handler        The pointer to the waypoint handler structure
+ * \param   waypoint_next           The next waypoint structure
+ */
+static void waypoint_handler_dubin_state_machine(mavlink_waypoint_handler_t* waypoint_handler, waypoint_local_struct_t* waypoint_next);
```

Comment:
The idea was to have navigation only having a waypoint to fly to and to do the logics of the waypoint in waypoint_handler. This function performs the switch from one Dubin's part to another (e.g. from DUBIN_CIRCLE_1 to DUBIN_STRAIGHT). 

I used the waypoint_handler structure to be consistent with what we did until now in most of the modules: to have only one argument in the function call to the main structure of the module.